retriever/endpointfinder/endpointfinder.go

```diff
@@ -30,22 +36,58 @@ type EndpointFinder struct {
 	minerInfoFetcher MinerInfoFetcher
 	h                host.Host
 	httpEndpoints    *lru.Cache[string, []peer.AddrInfo]
+	endpointErrors   *expirable.LRU[string, error]
+}
+
+const (
+	defaultLruSize         = 128
+	defaultErrorLruSize    = 128
+	defaultErrorLruTimeout = 5 * time.Minute
+)
+
+type EndpointFinderConfig struct {
+	LruSize         int
+	ErrorLruSize    int
+	ErrorLruTimeout time.Duration
+}
+
+func (cfg EndpointFinderConfig) lruSize() int {
+	if cfg.LruSize == 0 {
+		return defaultLruSize
+	}
+	return cfg.LruSize
+}
+
+func (cfg EndpointFinderConfig) errorLruSize() int {
+	if cfg.ErrorLruSize == 0 {
+		return defaultErrorLruSize
+	}
+	return cfg.ErrorLruSize
+}
+
+func (cfg EndpointFinderConfig) errorLruTimeout() time.Duration {
+	if cfg.ErrorLruTimeout == 0 {
+		return defaultErrorLruTimeout
+	}
+	return cfg.ErrorLruTimeout
 }
 
 // NewEndpointFinder returns a new instance of an EndpointFinder
-func NewEndpointFinder(minerInfoFetcher MinerInfoFetcher, h host.Host, size int) (*EndpointFinder, error) {
-	httpEndpoints, err := lru.New[string, []peer.AddrInfo](size)
+func NewEndpointFinder(cfg EndpointFinderConfig, minerInfoFetcher MinerInfoFetcher, h host.Host) (*EndpointFinder, error) {
```

Comment:
Instead of passing the config struct, it might be better to have option params to set non-default values

Reply:
yeah, that felt too heavy-weight but I guess I ended up with something that's probably going to be just as many lines of code but is a lesser option!